tests/e2e-playwright/specs/frontend/course-frontend.spec.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { expect, Page } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import { LessonPage } from '../../pages/frontend/lesson';
+import { CoursePage } from '../../pages/frontend/course';
+import { asAdmin, studentRole } from '../../helpers/context';
+import { Course, CourseMode, test } from './course-fixtures';
+import { disableGlobalLearningMode } from '../../helpers/api';
+import { text as lessonTextContent } from '../../helpers/lesson-templates';
+
+const { describe, use, beforeAll } = test;
+
+async function enroll( page: Page, course: Course ) {
+	const coursePage = new CoursePage( page );
+	await page.goto( course.link );
+	await coursePage.takeCourse.click( { force: true } );
+}
+
+const testMode = ( courseMode: CourseMode ) =>
+	describe( `Course Frontend (${ courseMode })`, () => {
+		use( { courseMode } );
+		use( studentRole() );
+
+		beforeAll( async ( { browser } ) => {
+			await asAdmin( { browser }, async ( { context } ) => {
+				await disableGlobalLearningMode( context );
+			} );
+		} );
+
+		test( 'Student enrolls in course', async ( { page, course } ) => {
```

Comment:
The idea of dividing tests into units is good for unit tests, but it is usually not much effective with e2e tests because when we are testing user journeys, the units are frequently repeated, which frequently leads us to very slow and expensive test suites, regardless our tech stack.

Example:
[Here we are looping all lessons and completing them to complete a course](https://github.com/Automattic/sensei/pull/6154/files#diff-60f61621bbad8af80a5a0226ef53eda506adeed6c58d1c2fb0a35d19769f77f5R65-R76)

So it is automatically covering the [previous test].(https://github.com/Automattic/sensei/pull/6154/files#diff-60f61621bbad8af80a5a0226ef53eda506adeed6c58d1c2fb0a35d19769f77f5R49-R62)

My suggestion is to really stick with the user journey idea, so we are really simulating a user that is doing a flow from the beginning to the end, without granularity, so ideally we should transform these 3 tests into a single one, where the user clicks on the taken course, completes all lessons, and checks if the course was completed.